### ISSUE TITLE
Unexpected shellcheck parse error for code that runs correctly and looks correct

### ISSUE THREAD

**Issue by xsdg** [koalaman/shellcheck#3475]
#### For bugs with existing features

- Rule Id (if any, e.g. SC1000): SC1009, SC1072, SC1073
- My shellcheck version (`shellcheck --version` or "online"): version: 0.11.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
geeqie_exe="$1"
completions_file="$2"

# Make sure input files exist.                                                                                                                                                                
if [ \! \( -x "$geeqie_exe" -a -e "$completions_file" \) ]; then
    echo "Missing input files"
    exit 2
fi
```

#### Here's what shellcheck currently says:

```
In ./build-aux/test-bash-completions.sh line 15:
if [ \! \( -x "$geeqie_exe" -a -e "$completions_file" \) ]; then
^-- SC1009 (info): The mentioned syntax error was in this if expression.
   ^-- SC1073 (error): Couldn't parse this test expression. Fix to allow more checks.
         ^-- SC1072 (error): Expected comparison operator (don't wrap commands in []/[[]]). Fix any mentioned problems and try again.

For more information:
  https://www.shellcheck.net/wiki/SC1072 -- Expected comparison operator (don...
  https://www.shellcheck.net/wiki/SC1073 -- Couldn't parse this test expressi...
  https://www.shellcheck.net/wiki/SC1009 -- The mentioned syntax error was in...
```

#### Here's what I wanted or expected to see:
This code runs as expected, and seems to be trivially correct

```sh
$(export geeqie_exe="/bin/ls"; export completions_file="/etc/passwd"; [ \! \( -x "$geeqie_exe" -a -e "$completions_file" \) ] && echo failcase || echo bypass)
bypass

$(export geeqie_exe="/bin/noexist"; export completions_file="/etc/passwd"; [ \! \( -x "$geeqie_exe" -a -e "$completions_file" \) ] && echo failcase || echo bypass)
failcase

$(export geeqie_exe="/bin/ls"; export completions_file="/etc/noexist"; [ \! \( -x "$geeqie_exe" -a -e "$completions_file" \) ] && echo failcase || echo bypass)
failcase

$(export geeqie_exe="/bin/noexist"; export completions_file="/etc/noexist"; [ \! \( -x "$geeqie_exe" -a -e "$completions_file" \) ] && echo failcase || echo bypass)
failcase
```

**Comment by wileyhy** [comment]
I can see you put some work into this BR/PR.

I'm sorry, but I feel it's worthwhile to at least mention how the same
functionality might more commonly be written.

if ! [ -x "${foo}" ] || ! [ -e "${bar}" ]
then
  echo bla
fi

Here's a quote from the POSIX spec addressing the ambiguous syntax of '-a',
'(', and ')':

  "The -a and -o binary primaries and the '(' and ')' operators have been
removed. (Many expressions using them were ambiguously defined by the
grammar depending on the specific expressions being evaluated.) Scripts
using these expressions should be converted...."
  - https://pubs.opengroup.org/onlinepubs/9799919799/utilities/test.html

Here's a few of the side effects:
- backslash escapes are absent so readability is improved;
- in xtrace, if one of the `[` commands fails, it would be clear to see
which one failed, which could save a step during any debugging process.

Wiley
